distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -780,9 +841,7 @@ Text.from_utf_8 bytes = Text_Utils.from_utf_8 bytes.to_array
 Text.utf_16 : Vector.Vector Integer
 Text.utf_16 = Vector.Vector (Text_Utils.get_chars this)
 
-## ADVANCED
```

Comment:
Shouldn't `from_utf_16` also use the decoding infrastructure? Currently it will not provide the same shape of warnings as `from_utf_8` and `from_bytes`.

Reply:
Or are they equivalent because creating a string from an array of any UTF-16 codepoints will always succeed? Feels like some of them can be invalid though.

Reply:
Java stores as UTF_16 so no conversion happening here. Could make it call via the Charset but it's an identity mapping between the scenes.

Reply:
For valid strings it is an identity mapping - true, but for [invalid strings which are representable in Java](https://stackoverflow.com/a/32271999/1296238) it will differ - passing thru the encoder will report errors and going directly as here will ignore them.

I think that given we handle errors in all the other cases, we should do so here too. Unless we explicitly want to be able to ignore these issues - as I can see some benefit in being able to construct these 'invalid' strings from invalid byte sequences (for example for testing purposes). But such mode should definitely be well documented and also IMO should be marked as `ADVANCED`.